internal/utils/version/version.go

```diff
@@ -0,0 +1,36 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package version
+
+import (
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+// CheckVersionMeetsConstraint returns true if the <version> meets the <constraint>.
+func CheckVersionMeetsConstraint(version, constraint string) (bool, error) {
```

Comment:
Is this something we can use to check the k8s version as well, say for specific features that require druid to be running on a certain version of k8s cluster?